plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java

```diff
@@ -939,7 +938,6 @@ private void updateProjectsWithProblems(List<String> addedProjectsUri) {
                 if (!projectConfig.getProblems().isEmpty()) {
                   try {
                     projectManager.update(projectConfig);
-                    eventService.publish(new ProjectUpdatedEvent(projectPath));
```

Comment:
AFAIK this is needed to re-update the project if the project is created before the java language servers started. @tolusha can you confirm it?

Reply:
No, we update all projects once the language server has finished starting.

Reply:
Sorry, not to re-update the project, but to redraw the project explorer tree.

Reply:
Anyway it causes a bug

Reply:
Can you say what scenario you see going wrong?

Reply:
I am not the author of this part of code, so I can't provide scenario when this line will be called. I mean that if we have `projectManager.update(projectConfig);`  and `eventService.publish(new ProjectUpdatedEvent(projectPath));` **AFAIK** is used to notify the client.
Let me rephrase my question: are you sure that we don't need this line?

Reply:
Or to be more precise: my testing indicates that we don't need this line.